l10n_it_invoices_data_communication/models/communication.py

```diff
@@ -0,0 +1,2557 @@
+# -*- coding: utf-8 -*-
+
+
+from openerp import api, fields, models, _
+from openerp.exceptions import ValidationError
+from openerp.osv import expression
+from lxml import etree
+import re
+
+
+NS_2 = 'http://ivaservizi.agenziaentrate.gov.it/docs/xsd/fatture/v2.0'
+VERSION = 'DAT20'
+NS_MAP = {
+    'ns2': NS_2,
+}
+etree.register_namespace("vi", NS_2)
+
+
+def format_decimal(value=0.0):
+    return "{:.2f}".format(value)
+
+
+def clear_xml_element(element):
+    if element.text:
+        return False
+    return all((clear_xml_element(e) for e in element.iterchildren()))
+
+
+def clear_xml(xml_root):
+    xml_root = etree.iterwalk(xml_root)
+    for dummy, xml_element in xml_root:
+        parent = xml_element.getparent()
+        if clear_xml_element(xml_element):
+            parent.remove(xml_element)
+
+
+def check_normalized_string(value):
+    normalized = True
+    if not value:
+        return normalized
+    if value != value.strip():
+        normalized = False
+    return normalized
+
+
+def _get_invoice_date_domain(date_start, date_end):
+    reg_date_domain = [
+        '&',
+        ('registration_date', '>=', date_start),
+        ('registration_date', '<=', date_end),
+    ]
+    date_inv_domain = [
```

Comment:
Questa ricerca per data registrazione o data fattura comporta che una fattura registrata in giugno e datata in maggio verrebbe spedita sia con la dichiarazione di maggio che con la dichiarazione di giugno, quindi doppiamente, che non è corretto.

Reply:
@sergiocorato ok, puoi proporre un'implementazione allora per questo `domain`?
Io non vedo alternative perché il campo `registration_date` potrebbe non esistere per le fatture validate prima dell'installazione di `account_invoice_entry_date`, come dicevo in https://github.com/OCA/l10n-italy/pull/1211#discussion_r281663999

Reply:
@SimoRubi usiamo `move_id.date` ?

Reply:
ho caricato una PR [qui](https://github.com/SimoRubi/l10n-italy/pull/10)